packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -45,6 +45,11 @@ var homepagePathname = homepagePath ? url.parse(homepagePath).pathname : '/';
 // Webpack uses `publicPath` to determine where the app is being served from.
 // It requires a trailing slash, or the file assets will get an incorrect path.
 var publicPath = ensureSlash(homepagePathname, true);
+// Not everybody wants a single-page app with client side routing without hash.
+// Sometimes, people would rather be able to deploy their app anywhere,
+// without having to specify a "homepage".
+// We can detect this when the publicPath is set to "./".
+var isPublicPathRelative = publicPath === './';
```

Comment:
I almost feel like this comment comes off a little demeaning to those trying to use client-side routing.
Could we make this friendlier and more terse?